src/Microsoft.Diagnostics.Runtime/src/Common/GCRoot.cs

```diff
@@ -174,7 +174,7 @@ public IEnumerable<GCRootPath> EnumerateGCRoots(ulong target, bool returnOnlyFul
             CancellationToken cancellationToken)
         {
             IClrRoot? root;
-            while ((root = queue.Take()) != null)
+            while ((root = queue.Take(cancellationToken)) != null)
```

Comment:
This will throw an OperationCancelledException, right?  This will be an unhandled exception on a worker thread which will tear down the process.  If you are adding `cancellationToken` to Add and Take you also need to wrap these with a catch/return.

Reply:
Good catch. In 0ccf44b37c7115c754bc1370a6949a261ac1246e I've removed the token from `Add` calls as it's unbounded, so won't block. To address the resulting warning, I've added a localised suppression using `#pragma warning disable`, as other suppression styles will suppress the diagnostic in the entire method.

Please also review:
- a8f8cbb9a549e0d935e3d344dac5b200e5c67316 which adds `BlockingCollection<>.CompleteAdding`
- 4e135291cbd3f7f4ed6f29bfbdd29ea09cfb683a which returns earlier in case of cancellation, which may also yield fewer results